summingbird-core/src/main/scala/com/twitter/summingbird/memory/Memory.scala

```diff
@@ -82,11 +116,11 @@ class Memory(implicit jobID: JobId = JobId("default.memory.jobId")) extends Plat
           case AlsoProducer(l, r) =>
             //Plan the first one, but ignore it
             val (left, leftM) = toStream(l, jamfs)
-            val (right, rightM) = toStream(r, leftM)
             // We need to force all of left to make sure any
             // side effects in write happen
-            lazy val lforcedEmpty = left.filter(_ => false)
-            (right.append(lforcedEmpty), rightM)
+            val lforcedEmpty = left.filter(_ => false)
+            val (right, rightM) = toStream(r, leftM)
```

Comment:
why do we have to revert this change? Calling `toStream(r, leftM)` first seems more correct to me, as does using the lazy concatenation (rather than the strict that we have reverted to).